File: @xoutput/client/src/rest/emulation/emulatedContollersClient.ts
import { ControllerInfo, CreateControllerRequest } from '@xoutput/api';
import { http } from '../http';

export const emulatedContollersClient = {
  getControllers() {
    return http.get<ControllerInfo[]>('/api/controllers');
  },
  createController(request: CreateControllerRequest) {
    return http.put<CreateControllerRequest>('/api/controllers', request);
  },
  startController(id: string) {
    return http.put(`/api/controllers/${id}/active`);
  },
  stopController(id: string) {
    return http.delete(`/api/controllers/${id}/active`);
  },
  deleteController(id: string) {
    return http.delete(`/api/controllers/${id}`);
  },
};

File: XOutput.Client/Rest/Emulation/EmulatedContollersClient.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace XOutput.Rest.Emulation
{
    public class EmulatedContollersClient : HttpJsonClient
    {
        public EmulatedContollersClient(IHttpClientProvider clientProvider) : base(clientProvider)
        {

        }

        public Task<IEnumerable<EmulatedControllerInfo>> GetControllers(int timeoutMillis = 1000)
        {
            return GetAsync<IEnumerable<EmulatedControllerInfo>>("/api/emulated/controllers", CreateToken(timeoutMillis));
        }

        public Task DeleteController(string id, int timeoutMillis = 1000)
        {
            return DeleteAsync($"/api/emulated/controllers/{id}", CreateToken(timeoutMillis));
        }
    }
}
